Core/Source/MVVM/ViewModel.swift

```diff
@@ -122,6 +122,32 @@ public protocol ViewModelBindingProvider {
 
     /// Returns a `ViewModel` for the given `Model` and context.
     func viewModel(for model: Model, context: Context) -> ViewModel
+
+    /// Returns the `SelectionViewModel` for given collection of models and a context. The default implementation
+    /// returns selection view model that works for a single model.
+    func selectionViewModel(for models: [Model], context: Context) -> SelectionViewModel?
+}
+
+extension ViewModelBindingProvider {
+    public func selectionViewModel(for models: [Model], context: Context) -> SelectionViewModel? {
+        if let firstModel = models.first, models.count == 1 {
+            return ViewModelSelectionShim(viewModels: [viewModel(for: firstModel, context: context)])
+        }
+        return nil
+    }
+}
+
+/// Represents a selection of one or more view models.
+public protocol SelectionViewModel {
```

Comment:
I wish this could inherit from `ViewModel` - but the initializer makes that trickier. thoughts? We *could* split out the non-initializer parts of `ViewModel` into a shared protocol, but that seems to be a bit too much overhead

Reply:
(thinking about it more, maybe we can incorporate into https://github.com/dropbox/pilot/pull/79 )